test/e2e/report_dynamic_input_data_test.go

```diff
@@ -163,3 +170,125 @@ func testReportsProduceData(t *testing.T, testReportingFramework *reportingframe
+		_, err := testReportingFramework.GetMeteringReport(reportName)
+		// `foundOnce` provides a latch that allows us to wait for report creation then exit on deletion
+		if err == nil {
+			foundOnce = true
+		}
+		if err != nil {
+			if foundOnce && apierrors.IsNotFound(err) {
+				return true, errors.New("report was deleted after being created")
+			}
+		}
+		// if we're exiting due to timeout, it's a fail
+		return false, nil
+	})
+	if err != nil {
+		assert.Contains(t, err.Error(), "was deleted", "report should have been deleted by retention period being reached")
+	}
+}
+
+func testReportIsNotDeletedWhenReportDependsOnIt(t *testing.T, testReportingFramework *reportingframework.ReportingFramework) {
+	testQueryName := "namespace-cpu-usage"
+	subReportName := "subreport-should-not-delete"
+	// cron schedule to run every minute
+	cronSchedule := &metering.ReportSchedule{
+		Period: metering.ReportPeriodCron,
+		Cron: &metering.ReportScheduleCron{
+			Expression: "*/1 * * * *",
+		},
+	}
+	var foundOnce = false
+	expiration := &metav1.Duration{Duration: 30.0 * time.Second}
+	// If a subReport is in dependency unmet state it won't be deleted and it can take awhile for import to happen.
+	// 4 minutes seems to be enough time for that setup, and the delete, to reliably happen.
+	waitTimeWhileCheckDeletion := 4 * time.Minute
+
+	subReport := testReportingFramework.NewSimpleReport(
+		subReportName,
+		testQueryName,
+		cronSchedule,
+		nil,
+		nil,
+		expiration,
+	)
+	t.Logf("creating subReport %s and waiting %s to finish", subReport.Name, waitTimeWhileCheckDeletion)
+
+	createErr := testReportingFramework.CreateMeteringReport(subReport)
+	assert.NoError(t, createErr, "creating subreport should produce no error")
+
+	newDefault := func(s string) *json.RawMessage {
+		v := json.RawMessage(s)
+		return &v
+	}
+
+	report := testhelpers.NewReport(
+		"report-depends-on-subreport",
+		testReportingFramework.Namespace,
+		testQueryName,
+		[]metering.ReportQueryInputValue{
+			metering.ReportQueryInputValue{
+				Name:  "NamespaceCPUUsageReportName",
+				Value: newDefault(`"` + subReportName + `"`),
+			},
+		},
+		nil,
+		nil,
+		metering.ReportStatus{},
+		cronSchedule,
+		false,
+		nil,
+	)
+
+	createErr = testReportingFramework.CreateMeteringReport(report)
+	assert.NoError(t, createErr, "creating report should produce no error")
+
+	err := wait.Poll(time.Second*5, waitTimeWhileCheckDeletion, func() (bool, error) {
+		_, err := testReportingFramework.GetMeteringReport(subReportName)
+		// `foundOnce` provides a latch that allows us to wait for subReport creation then exit on deletion
+		if err == nil {
+			foundOnce = true
+		}
+		if err != nil {
+			if foundOnce && apierrors.IsNotFound(err) {
+				return true, errors.New("subReport was deleted after being created")
+			}
+		}
+		// if we're exiting due to timeout, it's correct, because the subReport should not be deleted
+		return false, nil
```

Comment:
```go
_, err := testReportingFramework.GetMeteringReport(subReportName)
if apierrors.IsAlreadyExists(err) {
    foundOnce = true
    return false, nil
}
...

Reply:
returning false for the condition function on `Poll` means stop polling, not what we want here. 

Also, I don't find the api call more readable than just the check to see if there is no err directly, it's certainly slower.